Tasks/DeleteFiles/DeleteFiles.ts

```diff
@@ -0,0 +1,91 @@
+/// <reference path="../../definitions/vso-task-lib.d.ts" />
+
+import path = require('path');
+import os = require('os');
+var tl = require('vso-task-lib');
+
+// contents is a multiline input containing glob patterns
+var contents: string[] = tl.getDelimitedInput('Contents', '\n');
+var sourceFolder = tl.getPathInput('SourceFolder');
+
+// include filter
+var includeContents = [];
+
+// exclude filter
+var excludeContents = [];
+
+for (var i = 0; i < contents.length; i++) {
+    var pattern = contents[i].trim();
+    tl.debug('include content pattern: ' + pattern);
+    var realPattern = path.join(sourceFolder, pattern);
+    includeContents.push(realPattern);        
+}
+
+// enumerate all files
+var files = [];
+var allPaths = tl.find(sourceFolder);
+tl.debug('allPaths: ' + allPaths);
+if (allPaths.length === 0) {
+    tl.debug('source folder not found. nothing to delete.');
+}
+
+var allFiles: string[] = [];
+var allFolders: string[] = [];
+
+// folders should be deleted last
+for (var i = 0; i < allPaths.length; i++) {
+    tl.debug("checking for directory: " + allPaths[i]);
+    if (!tl.stats(allPaths[i]).isDirectory()) {
+        allFiles.push(allPaths[i]);
+    }
+    else {
+        allFolders.push(allPaths[i]);
+    }
+}
```

Comment:
I think folder should be deleted after all its files got delete.
not delete all files, then delete all folders. this seems weird.